source/dyaml/emitter.d

```diff
@@ -1098,52 +1107,107 @@ struct Emitter(Range, CharType) if (isOutputRange!(Range, CharType))
                                        scalar[index + 2].isSpace;
             }
 
-            with(analysis.flags)
+            with(ScalarAnalysis.AnalysisFlags)
             {
                 //Let's decide what styles are allowed.
-                allowFlowPlain = allowBlockPlain = allowSingleQuoted
-                               = allowDoubleQuoted = allowBlock = true;
+                analysis.flags |= allowFlowPlain | allowBlockPlain | allowSingleQuoted |
+                               allowDoubleQuoted | allowBlock;
 
                 //Leading and trailing whitespaces are bad for plain scalars.
                 if(leadingSpace || leadingBreak || trailingSpace || trailingBreak)
                 {
-                    allowFlowPlain = allowBlockPlain = false;
+                    analysis.flags &= ~(allowFlowPlain | allowBlockPlain);
```

Comment:
BitFlag should have this built-in, (e.g exclude...) so that the user could use it without typing any bitwise operation on it.